src/Toolkit/Toolkit/UI/Controls/FeatureForm/DatePickerFormInputView.cs

```diff
@@ -0,0 +1,130 @@
+﻿#if WPF
+using Esri.ArcGISRuntime.Mapping.FeatureForms;
+using Esri.ArcGISRuntime.Toolkit.Internal;
+using System.ComponentModel;
+
+namespace Esri.ArcGISRuntime.Toolkit.Primitives
+{
+    /// <summary>
+    /// Picker for the <see cref="DateTimePickerFormInput"/>.
+    /// </summary>
+    [TemplatePart(Name = "DatePicker", Type = typeof(DatePicker))]
+    public class DateTimePickerFormInputView : Control
+    {
+        private WeakEventListener<DateTimePickerFormInputView, INotifyPropertyChanged, object?, PropertyChangedEventArgs>? _elementPropertyChangedListener;
+        private DatePicker? _datePicker;
+        private TimePicker? _timePicker;
+
+        /// <summary>
+        /// Initializes an instance of the <see cref="DateTimePickerFormInputView"/> class.
+        /// </summary>
+        public DateTimePickerFormInputView()
+        {
+            DefaultStyleKey = typeof(DateTimePickerFormInputView);
+        }
+
+        /// <inheritdoc />
+        public override void OnApplyTemplate()
+        {
+            base.OnApplyTemplate();
+            _datePicker = GetTemplateChild("DatePicker") as DatePicker;
+            _timePicker = GetTemplateChild("TimePicker") as TimePicker;
+            if (_datePicker != null)
+            {
+                _datePicker.SelectedDateChanged += DatePicker_SelectedDateChanged;
+            }
+            if (_timePicker != null)
+            {
+                _timePicker.TimeChanged += TimePicker_TimeChanged;
+            }
+            ConfigurePickers();
+        }
+
+        private void TimePicker_TimeChanged(object? sender, EventArgs e) => UpdateValue();
+
+        private void DatePicker_SelectedDateChanged(object? sender, SelectionChangedEventArgs e) => UpdateValue();
+
+        private void UpdateValue()
+        { 
+            if (Element?.Input is DateTimePickerFormInput input && _datePicker != null)
+            {
+                var date = _datePicker.SelectedDate;
+                if(date.HasValue && input.IncludeTime && _timePicker != null && _timePicker.Time.HasValue)
+                {
+                    date = date.Value.Subtract(date.Value.TimeOfDay).Add(_timePicker.Time.Value);
```

Comment:
Optional: Another slightly-simpler-way of doing this, using [DateTime.Date](https://learn.microsoft.com/en-us/dotnet/api/system.datetime.date?view=netstandard-2.0):
```suggestion
                    date = date.Value.Date.Add(_timePicker.Time.Value);
```